code/game/objects/items/colorizers/colorizer.dm

```diff
@@ -0,0 +1,48 @@
+
+/obj/item/colorizer
+    name = "ERROR Colorizer"
+    desc = "This colorizer will apply a new set of colors to an item."
+    icon = 'icons/obj/crayons.dmi'
+    icon_state = "rainbowcan"
+
+    var/list/allowed_targets = list()
+    var/apply_icon = null
+    var/apply_icon_state = null
+    var/apply_item_state = null
+    var/apply_righthand_file = null
+    var/apply_lefthand_file = null
+
+
+/obj/item/colorizer/attack_self(mob/user)
+    var/obj/item/target_atom = user.get_inactive_held_item()
+    do_colorize(target_atom, user)
+    . = ..()
+
+/obj/item/colorizer/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+    do_colorize(target, user)
+    . = ..()
+
+/obj/item/colorizer/proc/do_colorize(atom/to_be_colored, mob/user)
+    if(!(to_be_colored && is_type_in_list(to_be_colored, allowed_targets)))
```

Comment:
```suggestion
    if(!to_be_colored ||! is_type_in_list(to_be_colored, allowed_targets))
```

Reply:
The double negative feels entirely unneccessary to me

Reply:
The provided suggestion is not equivalent in behavior.
![Discord_fb9HTJ8M4D](https://user-images.githubusercontent.com/5572280/144373589-375e9e22-e5fd-4cf2-a69d-f8d56f30ced2.png)

Reply:
if there is no item to be colored or the item that is to be colored isn't in typelist
functionally it's the same as negating their sum

Reply:
Actually if you are not passing any item here it should run silently and not display this message

Reply:
```suggestion
    if(!to_be_colored)
        return
    if(!is_type_in_list(to_be_colored, allowed_targets))
```